components/atoms/SoulCounter/useSoulCounter.tsx

```diff
@@ -1,34 +1,21 @@
-import { ChangeEvent, useReducer, useState } from "react";
-import { getRuneById } from "../../../constants/runes";
+import { ChangeEvent, useState } from "react";
+import { allRunes, getRuneById } from "../../../constants/runes";
 import { replaceRune } from "../../../utils/runeUtils";
 
 export type SoulCounterReturn = {
-  increment: (souls: number, id: number) => () => void;
-  decrease: (souls: number, id: number) => () => void;
-  total: number;
+  increase: (id: number) => () => void;
+  decrease: (id: number) => () => void;
   reset(): void;
   runeCount: InventoryRune[];
   setExactCount: (id: number) => (e: ChangeEvent<HTMLInputElement>) => void;
 };
 
 export default (): SoulCounterReturn => {
-  const [total, dispatch] = useReducer((state: number, action: any) => {
-    if (action.type === "raw") {
-      return action.value;
-    }
-    if (action === "reset") {
-      return 0;
-    }
-    if (state <= 0 && action <= 0) {
-      return 0;
-    }
-    return state + action;
-  }, 0);
+  const [runeCount, setCount] = useState<InventoryRune[]>(
+    allRunes.map((rune) => ({ ...rune, count: 0 }))
```

Comment:
There was a React warning regarding an uncontrolled input. Let me know how you feel about setting all of them to 0 on load @sergnio. If this works for you I think we can clean up some of the null checks inside the update functions.